### PR TITLE
Improve Pulumi error messages

### LINKED ISSUE #1623: Pulumi: misleading error messages
When running `dsh init`, if the `--name` parameter is not provided, the error message says that the `--deployment-name` parameter is not provided. This was the old name for that parameter. 

Another misleading error message

When attempting to use commands other then `dsh init` before having run `dsh init`, the error messages states that the `--deployment-name` parameter is missing. Providing the parameter correctly errors, since `--deployment-name` parameter is for `dsh init`, not for the other commands.

another misleading error

### PR DESCRIPTION
**Pull request by craddm**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).
- [ ] You have marked this pull request as a **draft** and added `'[WIP]'` to the title if needed (if you're not yet ready to merge).
- [ ] You have formatted your code using appropriate automated tools (for example `./tests/AutoFormat_Powershell.ps1 -TargetPath <path to file or directory>` for Powershell).

### :arrow_heading_up: Summary

Fixes incorrect error messages for the deployment `name` argument passed to `dsh init`
Adds an error when the backend has not been initialised before attempting to run commands that rely on it (e.g. `dsh admin list-users`)

Note - this PR also updates the relevant .devcontainer to Pulumi version `3.80.0`

### :closed_umbrella: Related issues

Closes #1623 

### :microscope: Tests

Tested locally:
better error

another better error

### CODE REVIEW

**Comment by jemrobinson** [comment]
It looks like this has been based off an older branch of `python-migration`. Possibly this is because you used a newer version of `develop`?

**Comment by craddm** [comment]
It shouldn't be - I updated my local copy of python-migration and merged it into this branch just before making the PR

Edit: ah, which makes it a mess and hard to see what I actually changed, right

**Comment by jemrobinson** [comment]
I've merged `develop` into the upstream `python-migration` ~and now this looks more sensible.~ hmm, no it doesn't. Can you `git reset --hard 283f0e93e` to drop your last merge commit?

**Review by jemrobinson — APPROVED** [review]
LGTM

File: .devcontainer/python/Dockerfile [hunk 1]
```diff
@@ -22,6 +21,7 @@ ARG TARGETARCH
 # Standard install method currently does not support ARM64
 # Use pip instead - https://github.com/Azure/azure-cli/issues/22875
 RUN pip3 install azure-cli==${AZURE_CLI_VERSION}
+RUN pip3 install hatch
 
 # Install Powershell
```
Comment (jemrobinson) [inline .devcontainer/python/Dockerfile]: Do we need Powershell in this devcontainer?

**Review by jemrobinson — APPROVED** [review]
LGTM

File: .devcontainer/python/Dockerfile (hunk 1 above)
Comment (craddm) [reply .devcontainer/python/Dockerfile]: No - fair point. Actually, do we need sphinx in there at the moment either? Or should I just leave it in for when we *do* need it? And what version of Azure CLI are you using at the moment?

Comment (jemrobinson) [reply .devcontainer/python/Dockerfile]: You're right that the requirements needed for building the docs aren't (currently) in the python-migration `pyproject.toml`. They probably ought to be added there once we've merged `python-migration` into develop, but until then I think it's fine to install the Sphinx `requirements.txt`.

**Pull request merged**